fili-core/src/main/java/com/yahoo/bard/webservice/data/time/StandardGranularityParser.java

```diff
@@ -5,39 +5,58 @@
 import com.yahoo.bard.webservice.druid.model.query.AllGranularity;
 import com.yahoo.bard.webservice.druid.model.query.Granularity;
 import com.yahoo.bard.webservice.util.GranularityParseException;
+import com.yahoo.bard.webservice.util.StreamUtils;
 
 import org.joda.time.DateTimeZone;
 
 import java.util.Arrays;
-import java.util.Collections;
 import java.util.Locale;
-import java.util.Map;
 import java.util.function.Function;
-import java.util.stream.Collectors;
+
+import javax.inject.Inject;
 
 /**
  * StandardGranularityParser implements a time grain dictionary, as well as factory methods to dynamically build zoned
  * time grains.
  */
 public class StandardGranularityParser implements GranularityParser {
 
-    private final Map<String, Granularity> namedGrains;
+    private final GranularityDictionary namedGranularities;
+
+    /**
+     * Constructor.
+     *
+     * @param dictionary  a dictionary containing names mapped to granularities
+     */
+    @Inject
+    public StandardGranularityParser(GranularityDictionary dictionary) {
+        namedGranularities = dictionary;
+    }
 
     /**
      * Constructor.
+     *
+     * Use a default grain map with the enum name()s and the all granularity.
      */
+    @Inject
     public StandardGranularityParser() {
-        namedGrains = Collections.unmodifiableMap(getGrainMap());
+        this(getDefaultGrainMap());
     }
 
     /**
      * This method loads default grains and can be extended to add customer grain extensions.
      *
      * @return A map of time grain api name to time grain instances.
      */
-    protected Map<String, Granularity> getGrainMap() {
-        Map<String, Granularity> result = Arrays.stream(DefaultTimeGrain.values())
-                .collect(Collectors.toMap(DefaultTimeGrain::name, Function.identity()));
+    public static GranularityDictionary getDefaultGrainMap() {
+        GranularityDictionary result = Arrays.stream(DefaultTimeGrain.values())
```

Comment:
If we expect this method to be called per-request (or at least in a request scope), we should memoize this method so that it doesn't need to re-build the same map over and over.